test/solidity/Facets/WhitelistManagerFacet.t.sol

```diff
@@ -0,0 +1,774 @@
+    WhitelistManagerFacet internal whitelistMgr;
+    AccessManagerFacet internal accessMgr;
+    Foo internal c1;
+    Foo internal c2;
+    Foo internal c3;
+
+    event AddressWhitelisted(address indexed whitelistedAddress);
+    event AddressRemoved(address indexed removedAddress);
+    event FunctionSelectorApprovalChanged(
+        bytes4 indexed functionSelector,
+        bool indexed approved
+    );
+
+    function setUp() public {
+        diamond = createDiamond(USER_DIAMOND_OWNER, USER_PAUSER);
+        whitelistMgr = new WhitelistManagerFacet();
+        c1 = new Foo();
+        c2 = new Foo();
+        c3 = new Foo();
+
+        bytes4[] memory functionSelectors = new bytes4[](10);
+        functionSelectors[0] = WhitelistManagerFacet.addToWhitelist.selector;
+        functionSelectors[1] = WhitelistManagerFacet
+            .removeFromWhitelist
+            .selector;
+        functionSelectors[2] = WhitelistManagerFacet
+            .batchAddToWhitelist
+            .selector;
+        functionSelectors[3] = WhitelistManagerFacet
+            .batchRemoveFromWhitelist
+            .selector;
+        functionSelectors[4] = WhitelistManagerFacet
+            .getWhitelistedAddresses
+            .selector;
+        functionSelectors[5] = WhitelistManagerFacet
+            .setFunctionApprovalBySelector
+            .selector;
+        functionSelectors[6] = WhitelistManagerFacet
+            .batchSetFunctionApprovalBySelector
+            .selector;
+        functionSelectors[7] = WhitelistManagerFacet
+            .isFunctionApproved
+            .selector;
+        functionSelectors[8] = WhitelistManagerFacet
+            .isAddressWhitelisted
+            .selector;
+        functionSelectors[9] = WhitelistManagerFacet
+            .getApprovedFunctionSelectors
+            .selector;
+
+        addFacet(diamond, address(whitelistMgr), functionSelectors);
+
+        // add AccessManagerFacet to be able to whitelist addresses for execution of protected functions
+        accessMgr = new AccessManagerFacet();
+
+        functionSelectors = new bytes4[](2);
+        functionSelectors[0] = accessMgr.setCanExecute.selector;
+        functionSelectors[1] = accessMgr.addressCanExecuteMethod.selector;
+        addFacet(diamond, address(accessMgr), functionSelectors);
+
+        accessMgr = AccessManagerFacet(address(diamond));
+        whitelistMgr = WhitelistManagerFacet(address(diamond));
+    }
+
+    function test_SucceedsIfOwnerAddsAddress() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        whitelistMgr.addToWhitelist(address(c1));
+        address[] memory approved = whitelistMgr.getWhitelistedAddresses();
+        assertEq(approved[0], address(c1));
+
+        vm.stopPrank();
+    }
+
+    function test_SucceedsIfOwnerRemovesAddress() public {
+        vm.startPrank(USER_DIAMOND_OWNER);
+
+        vm.expectEmit(true, true, true, true);
+        emit AddressWhitelisted(address(c1));
```

Comment:
You could call test_SucceedsIfOwnerAddsAddress() here and move the expectEmit into that test case